src/filter_finder.py

```diff
@@ -0,0 +1,208 @@
+						
+						features["raw"] = frame
+						
+						# get all masks and features for each filter
+						for key, (lower,upper) in self.filters.iteritems():
+							# create mask
+							masks[key] = cv2.inRange(hsv_frame, lower, upper)
+							# extract feature using mask
+							feature = cv2.bitwise_and(frame, frame, mask = masks[key])
+							
+							# write current filter info onto each filtered feature
+							lower_text = "lower: [{0},{1},{2}]".format(lower[0], lower[1], lower[2])
+							upper_text = "upper: [{0},{1},{2}]".format(upper[0],upper[1],upper[2])
+							cv2.putText(feature, lower_text, font_position, font, font_scale, font_color, font_thickness)
+							cv2.putText(feature, upper_text, (font_position[0], font_position[1]-30), font, font_scale, font_color, font_thickness)
+
+							features[key] = feature
+
+						# reset combined mask before recalculating it
+						# TODO: find a way around having to do this
+						masks["combined"] = 0
+						masks["combined"] = self.combine_masks(masks)
+						
+						features["combined"] = cv2.bitwise_and(frame, frame, mask = masks["combined"])
+						
+						# show all things
+						for key, feature in features.iteritems():
+							cv2.imshow(key, feature)
+						
+						# exit early if escape key pressed
+						k = cv2.waitKey(5) & 0xFF
+						if k == 27:
+							break
+
+						# use defined keyboard controls to update upper and lower bounds of chosen color range
+
+						# update lower bounds
+						for key, channel, sign, increment in self.lower_controls:
+							if k == ord(key):
+								lower_chosen[channel] = (lower_chosen[channel] + (sign * self.increments[increment])) % 256
+						# update upper bounds
+						for key, channel, sign, increment in self.upper_controls:
+							if k == ord(key):
+								upper_chosen[channel] = (upper_chosen[channel] + (sign * self.increments[increment])) % 256
+						
+						# keyboard controls to reset experimental color range
+						if k == ord('r'):
+							lower_chosen[:] = 0
+							upper_chosen[:] = 255
+						if k == ord('R'):
+							lower_chosen[:] = 0
+							upper_chosen[:] = 0
+
+						# advance frame
+						success, frame = cap.read()
+						
+					cv2.destroyAllWindows()
+					
+					# update values of this filter if desired
+					if (write):
+						self.filters[chosen][0], self.filters[chosen][1] = lower_chosen, upper_chosen
+					return (lower_chosen, upper_chosen)
+				# TODO: implement actual exception/error handling
+				else: print "The file provided could not be read. Make sure the given path leads to a video file."
+			else: 
+				print ("The desired filtering parameter \"{0}\" was not found "+
+				"in the parameter set:\r\n{1}").format(chosen, self.filters.keys())
+		else: print "The supplied filepath did not lead to a file."
+	
+	# return a set of predetermined filters based on the trials using Steven's skateboard
+	def get_default_filters(self):
+		return {"board":[np.array([82,39,80]), np.array([100,155,180])], 
```

Comment:
```suggestion
		return {"board": [np.array([82, 39, 80]), np.array([100, 155, 180])], 
```

Reply:
If I could have gotten VSCode to stop being broken, I would use it to refactor. NPP **bad**